els-back/app/Http/Controllers/User/AuthenticationController.php

```diff
@@ -58,19 +58,12 @@ public function logout(Request $request)
         return "success";
     }
 
-    public function getActivities($followed_ids) 
-    {
-        return Activity::whereIn('user_id',$followed_ids)->with('activityable','user')->orderBy('created_at','desc')->get();
-    }
 
     public function details(Request $request)
     {
         $user = $request->user()->load('learned_words.word','follows','followers','results');
 
-        $followed_ids = $user->follows()->pluck('id')->toArray();
-        $followed_ids[] = $user->id;
-
-        $user->activities = $this->getActivities($followed_ids);
+        $user->activities = Activity::whereIn('user_id',$user->scopeMineAndFollowingIds())->with('activityable','user')->orderBy('created_at','desc')->get();
```

Comment:
This is also used in user controller. Much better if you can create a scope/method as well so it can be reusable.
Note that business logic like this should be in model. The pupose of controller is calling the business logic like user->getActivities() while the model will process the logic behind getting activities

Cleaner and readable